docker/docker_client.go

```diff
@@ -428,15 +379,34 @@ func getAuth(ctx *types.SystemContext, registry string) (string, string, error)
 }
 
 type pingResponse struct {
-	WWWAuthenticate string
-	APIVersion      string
-	scheme          string
+	APIVersion string
+	scheme     string
+}
+
+func (c *dockerClient) saveChallenges(chs []challenge, u *url.URL) {
+	urlCopy := url.URL{
+		Path:   u.Path,
+		Host:   u.Host,
+		Scheme: u.Scheme,
+	}
+	normalizeURL(&urlCopy)
+	c.challenges[urlCopy.String()] = chs
+}
+
+func (c *dockerClient) getChallenges(req *http.Request) []challenge {
+	ping := url.URL{
+		Host:   req.URL.Host,
+		Scheme: req.URL.Scheme,
+		Path:   "/v2/",
```

Comment:
@mtrmac bottom line is, this is taken directly from docker (well, the way not the actual code). I believe if it works there, it'll work here as well. And again, this is removing a problematic request that we were making when setting up authentication to the registries.